core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -555,13 +560,11 @@ public void start(ClientStreamListener listener) {
         serverStream.setListener(listener);
 
         synchronized (InProcessTransport.this) {
-          ServerStreamListener serverStreamListener = serverTransportListener.streamCreated(
-              serverStream, method.getFullMethodName(), headers);
-          clientStream.setListener(serverStreamListener);
           streams.add(InProcessTransport.InProcessStream.this);
           if (streams.size() == 1) {
             clientTransportListener.transportInUse(true);
           }
+          serverTransportListener.streamCreated(serverStream, method.getFullMethodName(), headers);
```

Comment:
It looks like `streamCreated()` still returns `serverStreamListener`. Is it possible to make it `void`? I could imagine that maybe tests become more difficult, but normal code shouldn't use the return value at all.